lib/cartopy/mpl/geoaxes.py

```diff
@@ -753,6 +753,20 @@ def add_geometries(self, geoms, crs, **kwargs):
 
         """
         styler = kwargs.pop('styler', None)
```

Comment:
What about updating the `styler` here? You can pass a callable that updates the color that way. You might be able to put the `isinstance` check in that way and it would act on each geometry then.

Reply:
OK. But maybe it should be clarified that we only do styler update for LineString when 
(1) styler is not set (`None`); or
(2) `facecolor` is not set, otherwise we should assume the user wants to fill the LineString by explicitly specifying `facecolor`. 

And, what if user passes a callable to styler?

Reply:
I agree with all of that. If a user passes a callable for styler then this should have no effect. You could probably add an `if callable(styler): return styler` in there somewhere.

This should also get some tests to make sure it is covering those cases you mentioned and that we are taking the right paths.